app/lib/big_query/request_event.rb

```diff
@@ -0,0 +1,47 @@
+module BigQuery
+  class RequestEvent
+    def initialize
+      @event_hash = {
+        environment: ENV["RAILS_ENV"],
+        occurred_at: Time.zone.now.iso8601,
+        event_type: "web_request",
+      }
+      yield self if block_given?
+    end
+
+    delegate :as_json, to: :event_hash
+
+    def with_request_details(request)
+      event_hash.merge!(
+        request_uuid: request.uuid,
+        request_path: request.path,
+        request_method: request.method,
+        request_user_agent: request.user_agent,
+        request_query: query_to_kv_pairs(request.query_string),
+        request_referer: request.referer,
+      )
```

Comment:
For the other services we've also been adding `anonymised_user_agent_and_ip` for ... well ... an anonymised version of the user agent and IP. I know this is less useful for the API, but it could still be interesting and it would be more consistent. Worth adding?

Reply:
Can do

Reply:
I've updated to include that now @misaka if you could have another 👀

Reply:
👍